pkg/engine/turn/modify.go

```diff
@@ -1,48 +1,64 @@
 package turn
 
 import (
+	"sort"
 	"github.com/simimpact/srsim/pkg/engine/event"
 	"github.com/simimpact/srsim/pkg/engine/info"
 	"github.com/simimpact/srsim/pkg/key"
 )
 
+// SetGauge sets the gauge of the target as detailed in the input ModifyAttribute struct.
+// 1. Update gauge of target
+// 2. Move target to top of order
+// 3. Sort target down based on AV. In the event of tie, this target should be at top of order.
+//		If there is an active turn and it is not this target, this target should go below the active
+//		turn (so index 1 instead of 0 when 0 gauge/AV)
+// 4. Emit GaugeChangeEvent
 func (mgr *manager) SetGauge(data info.ModifyAttribute) error {
-	prev := mgr.target(data.Target).gauge
+
+	previousGauge := mgr.target(data.Target).gauge
 	mgr.target(data.Target).gauge = data.Amount
 
-	if mgr.target(data.Target).gauge == prev {
+	// if there's no change to Gauge, exit early
+	if mgr.target(data.Target).gauge == previousGauge {
 		return nil
 	}
```

Comment:
best to minimize this code to only call `mgr.target(data.Target)` once since that is a for loop over all targets every time it is called